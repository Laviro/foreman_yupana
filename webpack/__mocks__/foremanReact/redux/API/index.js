export const get = data => ({ type: 'get-some-type', ...data });
export const put = data => ({ type: 'put-some-type', ...data });
export const post = data => ({ type: 'post-some-type', ...data });
export const patch = data => ({ type: 'patch-some-type', ...data });

export const API = {
  get: jest.fn(),
  put: jest.fn(),
  post: jest.fn(),
  delete: jest.fn(),
  patch: jest.fn(),
};
