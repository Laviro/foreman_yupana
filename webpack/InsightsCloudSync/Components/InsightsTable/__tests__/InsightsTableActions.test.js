import { testActionSnapshotWithFixtures } from '@theforeman/test';
import API from 'foremanReact/redux/API';
import {
  fetchInsights,
  setSelectAllAlert,
  selectByIds,
  setSelectAll,
  selectAll,
  clearAllSelection,
} from '../InsightsTableActions';
import { hits } from './fixtures';

jest.mock('foremanReact/redux/API', () => jest.fn());
API.get = jest.fn(({ handleSuccess, ...action }) => {
  handleSuccess({ data: { hits } });
  return { type: 'get', ...action };
});

const runWithGetState = (state, action, params) => dispatch => {
  const getState = () => ({
    router: {
      location: {
        query: {
          page: '1',
          per_page: '7',
          search: '',
          sort_by: '',
          sort_order: '',
          select_all: 'true',
        },
      },
    },
  });
  action(params)(dispatch, getState);
};

const fixtures = {
  'should fetchInsights': () =>
    runWithGetState({}, fetchInsights, { page: 2, perPage: 7 }),
  'should setSelectAllAlert true': () => setSelectAllAlert(true),
  'should selectByIds': () => selectByIds({ 1: true, 5: true }),
  'should setSelectAll false': () => setSelectAll(false),
  'should selectAll': () => selectAll(),
  'should clearAllSelection': () => clearAllSelection(),
};

describe('insights table actions', () =>
  testActionSnapshotWithFixtures(fixtures));
