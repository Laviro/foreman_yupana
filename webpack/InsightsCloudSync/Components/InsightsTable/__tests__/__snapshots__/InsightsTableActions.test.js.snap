// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insights table actions should clearAllSelection 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "selectedIds": Object {},
      },
      "type": "INSIGHTS_SET_SELECTED_IDS",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "showSelectAllAlert": false,
      },
      "type": "INSIGHTS_SET_SELECT_ALL_ALERT",
    },
  ],
  Array [
    [Function],
  ],
]
`;

exports[`insights table actions should fetchInsights 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "selectedIds": Object {
          "16": true,
          "17": true,
        },
      },
      "type": "INSIGHTS_SET_SELECTED_IDS",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "showSelectAllAlert": true,
      },
      "type": "INSIGHTS_SET_SELECT_ALL_ALERT",
    },
  ],
  Array [
    [Function],
  ],
  Array [
    Object {
      "key": "INSIGHTS_HITS",
      "params": Object {
        "order": " ",
        "page": 2,
        "per_page": 7,
        "search": "",
      },
      "type": "get",
      "url": "/insights_cloud/hits",
    },
  ],
  Array [
    Object {
      "payload": Object {
        "args": Array [
          Object {
            "pathname": "/foreman_rh_cloud/insights_cloud",
            "search": "?page=2&per_page=7&search=&sort_by=&sort_order=&select_all=true",
          },
        ],
        "method": "push",
      },
      "type": "@@router/CALL_HISTORY_METHOD",
    },
  ],
]
`;

exports[`insights table actions should selectAll 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    Object {
      "payload": Object {
        "isAllSelected": true,
      },
      "type": "INSIGHTS_SET_SELECT_ALL",
    },
  ],
]
`;

exports[`insights table actions should selectByIds 1`] = `
Object {
  "payload": Object {
    "selectedIds": Object {
      "1": true,
      "5": true,
    },
  },
  "type": "INSIGHTS_SET_SELECTED_IDS",
}
`;

exports[`insights table actions should setSelectAll false 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    Object {
      "payload": Object {
        "isAllSelected": false,
      },
      "type": "INSIGHTS_SET_SELECT_ALL",
    },
  ],
]
`;

exports[`insights table actions should setSelectAllAlert true 1`] = `
Object {
  "payload": Object {
    "showSelectAllAlert": true,
  },
  "type": "INSIGHTS_SET_SELECT_ALL_ALERT",
}
`;
